apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.TargetedSessionPool.cs

```diff
@@ -179,14 +178,14 @@ private async Task<PooledSession> AcquireSessionImplAsync(TransactionOptions tra
                 if (transactionMode == ModeOneofCase.ReadWrite)
                 {
                     Interlocked.Increment(ref _rwTransactionRequests);
-                    if (session.TransactionMode == transactionMode)
+                    if (session.TransactionOptions.ModeCase == transactionMode)
                     {
                         Interlocked.Increment(ref _rwTransactionRequestsPrewarmed);
                     }
                 }
 
                 // If we've already got the right transaction mode, we're done.
-                if (session.TransactionMode == transactionMode)
+                if (session.TransactionOptions.ModeCase == transactionMode)
```

Comment:
Is that actually the case? If we've been given transaction options for a read-only transaction, but with a different read_timestamp to the one in the pool for example, we can't just use the same pooled session, presumably?